crates/ide_assists/src/handlers/add_missing_match_arms.rs

```diff
@@ -178,6 +178,19 @@ pub(crate) fn add_missing_match_arms(acc: &mut Assists, ctx: &AssistContext) ->
                             None => Cursor::Before(first_new_arm.syntax()),
                         };
                     let snippet = render_snippet(cap, new_match_arm_list.syntax(), cursor);
+                    // remove the second last line if it only contains trailing whitespace
+                    let lines = snippet.lines().collect_vec();
+                    let snippet = lines
+                        .iter()
+                        .enumerate()
+                        .filter_map(|(index, &line)| {
+                            if index + 2 == lines.len() && line.trim().is_empty() {
+                                return None;
+                            }
+                            return Some(line);
+                        })
+                        .join("\n");
+
```

Comment:
I'm not sure if this is the right way to remove the second last line if it only contains trailing white space

Reply:
Ye this looks odd, I think it should be fine to just leave the empty line be for now, it should be at the end after insertion still right?

Reply:
That was my thought too but leaving the empty line would fail the tidy test for including trailing whitespace in the test case.

Reply:
Ah, then I think we can just adjust the test case to have the cursor at the end of the last match arm instead of a new line.

Reply:
fixed in 1f3da38d94ddbc728e6d5c6eef69166406329b97, I kept the comment unchanged at https://github.com/rust-analyzer/rust-analyzer/pull/10759/files#diff-b4a4988c91896651971cb035f7b6da4f06360410ef3c7da31b34d1c622c40523R200-R203 .